### ISSUE TITLE
Allow a user to block an instance

### ISSUE THREAD

**Issue by wiki-me** [LemmyNet/lemmy#2397]
For front end issues, use [lemmy-ui](https://github.com/LemmyNet/lemmy-ui)

### Is your proposal related to a problem?

When looking at the "all" feed and even sorting by "top week" , it is basically all communist propaganda/advocacy ,obviously a lot of people are not interested in that (And for some people who come or have family from ex communist countries it can even be emotionally aversive ). you can block communities but there are a lot of them and they can keep popping up.

### Describe the solution you'd like

Give users the option to block a certain instance (e.g. lemmygrad).

### Describe alternatives you've considered

None

### Additional context

Dessalines saw [this](https://lemmy.ml/post/353544/comment/226144) post where i criticised lemmy and he asked i will open issues.

### updates

There is a [workaround script](https://github.com/LemmyNet/lemmy/issues/2397#issuecomment-1658761077).

One of the maintainers [said](https://github.com/LemmyNet/lemmy/issues/2397#issuecomment-1610973647) he is willing to merge this if somebody will implement it

**Comment by Nutomic** [comment]
Would it be an option for you to switch to another instance which blocks lemmygrad.ml? For example https://sopuli.xyz/ or https://beehaw.org/

**Comment by dessalines** [comment]
This may be a non-issue, because I've added an easy way to block communities from that community page (you could already block them by going into your user settings). So you could just block communities from instances you'd rather not see, as they show up in your feed.

I kind of agree with nutomic tho, that if you don't want to see communism in your feed, you're better off signing up for an explicitly anti-communist instance.

**Comment by wiki-me** [comment]
Still it's a hassle, with multiple communities you have to block, not to mention new communities will pop up and you will have to block them also, it seems like too much of a hassle and a waste of time.

regarding switching instances, I don't want to make this discussion too specific to my personal situation, but i think most social media use today tend to create a [filter bubble](https://en.wikipedia.org/wiki/Filter_bubble) which prevent people from the beneficial effects of testing their beliefs, which is already a risk because appparently people have a [tendency](https://en.wikipedia.org/wiki/Confirmation_bias) to do that too little. So unless that instance does something that is really not something acceptable (stuff like pokemon porn or gore) , i would rather decide for myself which instances should be blocked.

**Comment by finnim** [comment]
Hello, I wanted to add to this now that I'm considering hosting a lemmy instance.

Here's my use case: 
- Allowing users to suspend remote instances reduces drama, reduces pressure on the moderation team and increases federation.

Say I'm hosting an instance which federates with other instances focused around controversial topics such as atheism, X or Y-wing politics, cryptocurrencies, AI generated content, etc...

Without giving users the ability to ban whole instances, small but very vocal minorities could engage in a toxic dynamic of trying to capture power by pressuring the staff to ban x, y or z content they disagree with. This could indeed lead to a more fragmented fediverse as inevitably some instances will cave.

It is much easier for staff and also beneficial for federation if admin can simply instruct their users to ban any remote instances they don't like instead of causing drama. 

To give a more relevant example, if I want to create an anime related instance it is very likely that some people will be bothered by lemmygrad. Since the instance is anime focused either I or other moderators could feel the need to suspend that instance in order to avoid toxic drama that could arise around controversial topics... however this is in not in my interest and neither is it in the interest of lemmygrad.

A) Lemmygrad obviously doesn't want to be suspended because that reduces its usefulness
B) I don't want to open the door to having to bother about remote instance moderation or dealing with people sending in countless reports because they disagree with something that's controversial

It is easier for me to say: we don't ban remote instances for being controversial unless they promote actively harmful content. If you have a problem with a remote instances just block it yourself and carry on.

Allowing users to suspend whole instances is simpler than having to hunt down individual remote communities and is a solution that benefits the controversial instance by not being blocked, the staff of the local instance by not having drama and having to police controversial but non-harmful topics on remote instances and finally it also benefits the user by just allowing them to block instances they don't like.

This is my personal opinion, but instance admins whether on mastodon or lemmy should very rarely ban whole remote instances (that breaks federation and causes a lot of problems). It's much better for instance admins to simply block instances that are actually harmful/illegal/really bad and tell their users to kindly shut up about about controversial stuff and simply block it themselves if they don't like it.

**Comment by ericjmorey** [comment]
The implementation of this feature should include a log and reporting accessable to instance admins of when a blocking action was taken and the current state of which instances are being blocked by which accounts.

**Comment by Atulin** [comment]
Seeing how the maintainers of Lemmy run lemmygrad, i don't see this feature ever being implemented since everybody would use it to block lemmygrad immediately.

**Comment by ShibePatrol** [comment]
I'd like to add my voice to this. Much of the controversy regarding defederation would be instantly resolved by adding this feature. It would improve the platform with absolutely no downsides.

**Comment by jenkshields** [comment]
I have another relevant use case for this - as more and more people are moving to lemmy and kbin instances, my all feed is filled with nsfw content from one instance. I block each community as I see it, but every day there's at least 2-3 new ones on the feed cropping up - it'd be much easier to be able to block the instance.

This is a case where defederation isn't a particularly useful answer - nsfw content on a feed is much more a user preference than, for example, simply moving to a server that defeds lemmygrad.

Simply filtering all nsfw content out isn't ideal either - there's no distinction between porn and other kinds of nsfw content i don't mind engaging with - especially with the number of apps that allow blurring of nsfw content now.

**Comment by Nutomic** [comment]
I would definitely merge this if someone takes the time to implement it.

**Comment by elishamint** [comment]
There is also suggestions to have a "soft defederation", which could actually work by just giving an admin the ability to insert an instance block in every user's blacklist. The user could still unblock if they wish. That would be a sensible way to deal with most of the future defederation debates but still keep the open spirit of federation.

There's another request with a "hide instance from 'all' feed" in its list: https://github.com/LemmyNet/lemmy/issues/3255

**Comment by csm10495** [comment]
An alternative approach could be to just turn the current community blocking from list of communities to list of community regexes. It would also need the ui to allow you to add communities that don't necessarily exist (at this point it validates them) since they would be regexes. 

That way you could block `siteidontlike.net/.*`

**Comment by Nutomic** [comment]
We already have a db table `instance` which is referenced by community and person rows. So there needs to be a new table `instance_block` similar to existing `person_block` table. Then these blocks need to be filtered in SQL queries inside `crates/db_views`. There are no regexes involved.

**Comment by TheSpaghettiProgrammer** [comment]
Just wanted to add yet another use case here. My feed has been getting more and more foreign language instances. Being able to block an entire instance in this case would allow me to remove the foreign language instances fairly easily to clean up my feed.

I like the idea of an instance_block table.

**Comment by trymeouteh** [comment]
Would like to see a setting for a user to block instances just as there are settings for the user to block users and communities

**Comment by K4LCIFER** [comment]
One thing that I'm wondering about is if a user blocks an instance, would that also hide users from that instance that may be commenting on posts? I, personally, would just like an option to block an instance's communities from showing up in my All feed.

**Comment by Nutomic** [comment]
My PR only hides communities from that instance, but not users.

**Comment by jfryton** [comment]
Allowing users to block instances from appearing in All would be very useful, especially for instances that avoid defederation. Ideally users would be able to manage which instances appear themselves (not necessarily blocking comments from their users elsewhere, but any communities from those instances).

**Comment by DraconicNEO** [comment]
> My PR only hides communities from that instance, but not users.

That seems like an issue as many people want instance blocking not just for filtering out communities but also users as well as certain instances may have large amounts of problematic users but in which the home instance admin has chosen not to defederate from them.

**Comment by Die4Ever** [comment]
> > My PR only hides communities from that instance, but not users.
> 
> That seems like an issue as many people want instance blocking not just for filtering out communities but also users as well as certain instances may have large amounts of problematic users but in which the home instance admin has chosen not to defederate from them.

I think ideally we would have 2 different options

* block communities from instance X
* block users from instance X

I could also see users being split up into posts from instance X vs comments from instance X but I think that might be overkill

**Comment by wiki-me** [comment]
I opened this issue , blocking users isn't really what i meant, i suggest opening a new issue if it already does not exist (something like "allow blocking users from specific instance"), the discussion here is not very relevant to the new request and i don't think people should feel obliged to read this entire discussion. You can always link to it here for the people following this issue.
